src/main/java/com/gtnewhorizons/gravisuiteneo/mixins/MixinItemVajra.java

```diff
@@ -52,4 +52,51 @@ public int getHarvestLevel(ItemStack stack, String toolClass) {
             @SuppressWarnings("rawtypes") List tooltip, boolean advancedTooltips, CallbackInfo ci) {
         tooltip.add(EnumChatFormatting.AQUA + StatCollector.translateToLocal("message.vajra.clickRightForSilk"));
     }
+
+    // The vajra onItemUse has two branches for figuring out what a block drops
+    // depending on whether or not the block canSilkHarvest. Then the branches come back together and set the block
+    // to air.
+    // the fixCallOrder redirect handles everything, so there is no need for these calls to do anything.
+    @Redirect(
+            method = "onItemUse",
+            at = @At(
+                    value = "INVOKE",
+                    ordinal = 0,
+                    target = "Lnet/minecraft/block/Block;onBlockHarvested(Lnet/minecraft/world/World;IIIILnet/minecraft/entity/player/EntityPlayer;)V"))
+    private void onBlockHarvestToNoOp(Block block, World world, int x, int y, int z, int meta, EntityPlayer player) {
+        return;
+    }
+
+    @Redirect(
+            method = "onItemUse",
+            at = @At(value = "INVOKE", target = "Lnet/minecraft/world/World;setBlockToAir(III)Z"))
+    private boolean setBlockToAirToNoOp(World world, int x, int y, int z) {
+        return true;
+    }
+
+    // harvestBlock would be called first by the vajra, so we redirect to the proper chain of calls
+    @Redirect(
+            method = "onItemUse",
+            at = @At(
+                    value = "INVOKE",
+                    target = "Lnet/minecraft/block/Block;harvestBlock(Lnet/minecraft/world/World;Lnet/minecraft/entity/player/EntityPlayer;IIII)V"))
+    private void fixCallOrder(Block block, World world, EntityPlayer player, int x, int y, int z, int meta) {
+        block.onBlockHarvested(world, x, y, z, meta, player);
+        world.setBlockToAir(x, y, z);
+        block.harvestBlock(world, player, x, y, z, meta);
+    }
+
+    // This one makes sure that if we're mining a block that canSilkHarvest it still gets set to air, since we yeeted
+    // the original setBlockToAir call above
+    @Inject(
+            method = "onItemUse",
+            at = @At(
+                    value = "INVOKE",
+                    shift = At.Shift.AFTER,
```

Comment:
Unneccessary shift because this injection does not change the `dropFlag` variable, nor is it affected by it.

Reply:
I wanted to have the injection happen after the assignment, because I thought it would look nicer if anybody ever read the decompiled mixin'd class. However, as it turns out, even with this shift the call still ends up before the assignment. Deleted.